meilisearch-release-changelog/v1.4.0.md

```diff
@@ -0,0 +1,89 @@
+
+Some SDKs might not include all new features—consult the project repository for detailed information. Is a feature you need missing from your chosen SDK? Create an issue letting us know you need it, or, for open-source karma points, open a PR implementing it (we'll love you for that ❤️).
+
+# New features and improvements 🔥
+
+## Customize text separators
+
+The Meilisearch word segmentation uses a default list of separators to segment words. This default list might not sometimes fit your use case: the `nonSeparatorTokens`/`separatorTokens`` setting allows you to remove/add some tokens from/to the default list of separators.
+
+To remove tokens:
+
+```bash
+curl \
+  -X PUT 'http://localhost:7700/indexes/articles/settings/non-separator-tokens' \
+  -H 'Content-Type: application/json'  \
+  --data-binary '["@", "#", "&"]'
+```
+
+To add tokens:
+```bash
+curl \
+  -X PUT 'http://localhost:7700/indexes/articles/settings/separator-tokens' \
+  -H 'Content-Type: application/json'  \
+  --data-binary '["&sect;", "&sep"]'
+```
+
+Done by @ManyTheFish in #3946
+
+## Load a user-defined dictionary
+
+The Meilisearch word segmentation relies on separators and language-based dictionaries to segment words. This segmentation could be more accurate on technical or use-case-specific vocabulary (like "G/Box" to say "Gear Box") or proper nouns (like "J. R. R." when parsing "J. R. R. Tolkien"). The `dictionary` setting improves this accuracy by defining a list of words that would be segmented as described in the list.
+
+```bash
+curl \
+  -X PUT 'http://localhost:7700/indexes/articles/settings/dictionary' \
+  -H 'Content-Type: application/json'  \
+  --data-binary '["J. R. R.", "J.R.R."]'
+```
+
+These feature synergies well with the `stopWords` or the `synonyms` settings allowing to segment words and correctly retrieve what you expect:
+
+```json
+{
+    "dictionary": ["J. R. R.", "J.R.R."],
+    "synonyms": {
+            "J.R.R.": ["jrr", "J. R. R."],
+            "J. R. R.": ["jrr", "J.R.R."],
+            "jrr": ["J.R.R.", "J. R. R."],
+    }
+}
+```
+
+Done by @ManyTheFish in #3946
+
+## Other improvements
+TBD
+
+# Fixes 🐞
+
+* Fix display of synonyms (#3946) @ManyTheFish
+
+# Misc
+
+* Dependencies upgrade
+  * Upgrade dependencies of the CI (#3871, #3969 and #3968)
+  * Upgrade Meilisearch dependencies (#3987) @ManyTheFish
+* CIs and tests
+  * Fix benchmarks CI (#3963) @irevoire
+  * Removed unnecessary borrow call that failed nightly tests (#3990) @JannisK89
+  * Expose a Puffin server to profile the indexing process (#3913) @Kerollmops
+  * Update and fix the Test Suite CI (#3918) @Kerollmops
+  * Improve test suite CI for manual trigger events (#3989) @curquiza
+* Documentation
+  * Update README.md (#3895 and #3932) @ferdi05 & @Strift
+* Misc
```

Comment:
Seems unnecessary to have sublists here, especially as the last one is a repetition of the section title (that is, `# Misc`).

```suggestion
  * Upgrade Meilisearch dependencies (#3987) @ManyTheFish
  * Upgrade CI dependencies (#3871, #3969 and #3968)
  * Fix CI benchmarks (#3963) @irevoire
  * Removed `borrow` call causing failed nightly tests (#3990) @JannisK89
  * Expose Puffin server to profile the indexing process (#3913) @Kerollmops
  * Update and fix the CI Test Suite (#3918) @Kerollmops
  * Improve CI test suite for manual trigger events (#3989) @curquiza
  * Update README.md (#3895 and #3932) @ferdi05 & @Strift
```